src/partition.rs

```diff
@@ -21,15 +21,60 @@ use crate::plane::*;
 use crate::predict::*;
 use crate::util::*;
 
-pub const NONE_FRAME: usize = 8;
-pub const INTRA_FRAME: usize = 0;
-pub const LAST_FRAME: usize = 1;
-pub const LAST2_FRAME: usize = 2;
-pub const LAST3_FRAME: usize = 3;
-pub const GOLDEN_FRAME: usize = 4;
-pub const BWDREF_FRAME: usize = 5;
-pub const ALTREF2_FRAME: usize = 6;
-pub const ALTREF_FRAME: usize = 7;
+// LAST_FRAME through ALTREF_FRAME correspond to slots 0-6.
+#[derive(PartialEq, Copy, Clone)]
+pub enum RefType {
+  NONE_FRAME = 8,
```

Comment:
Is there some convention to put the none value at the beginning ? Otherwise I'd put it at the end to keep things in order. Also, are there situations where we know that NONE_FRAME cannot occur? If so I'd remove it from the enum and use `Option<RefType>` where we need it.

Reply:
It was that way in libaom. I will switch it. Yes, there are a lot of situations where both NONE_FRAME and INTRA_FRAME cannot occur (basically everywhere to_index() is used). I'm going to actually make a separate enum type for those cases. Then maybe we can drop RefType entirely.